fedimg/services/ec2.py

```diff
@@ -717,6 +717,44 @@ def upload(self, compose_meta):
                         image.id, self.build_name, self.virt_type,
                         self.vol_type, ami['region']))
 
+                # Make the snapshot for the image public.
+                is_snapshot_public = False
+                snapshot = None
+
+                alt_ami = alt_driver.get_image(image.id)
+                blk_device_mapping = alt_ami.extra['block_device_mapping']
+                if len(blk_device_mapping) == 1:
+                    snapshot_id = blk_device_mapping[0]['ebs']['snapshot_id']
+                    snapshot_obj = type('', (), {})()
+                    snapshot_obj.id = snapshot_id
+                    snapshot = alt_driver.list_snapshots(snapshot=snapshot_obj)
+
+                if snapshot is not None:
```

Comment:
[From the horse's mouth](https://www.python.org/dev/peps/pep-0008/#programming-recommendations)

Comparisons to singletons like None should always be done with is or is not , never the equality operators.

Also, beware of writing if x when you really mean if x is not None -- e.g. when testing whether a variable or argument that defaults to None was set to some other value. The other value might have a type (such as a container) that could be false in a boolean context!